src/views/Editor.vue

```diff
@@ -529,6 +538,28 @@ export default {
 				}
 			})
 		},
+		getError(error,apiCall,datasetId) {
```

Comment:
is this function a full copy?

Reply:
Slight change as Dataset handles both errors (backend and timeout etc). However for Editor we had to apply check

Reply:
modified existing method to have common body of both methods. and exported into a new file